Creation of cat.md for pytorch (#5405)
Codecademy/docs: content/pytorch/concepts/tensors/terms/cat/cat.md

---
Title: '.cat()'
Description: 'Concatenates two or more tensors in the same dimension.'
Subjects:
  - 'AI'
  - 'Data Science'
Tags:
  - 'AI'
  - 'Deep Learning'
  - 'Functions'
  - 'Machine Learning'
CatalogContent:
  - 'intro-to-py-torch-and-neural-networks'
  - 'py-torch-for-classification'
---

The **`.cat()`** function in PyTorch is used for concatenating two or more tensors in the same dimension. The tensors must have the same shape in all dimensions except the one in which they are concatenated.

## Syntax

```pseudo

> torch.cat(tensors, dim=0)

Reviewer: Can we also mention the out parameter here? I think it would be helpful. Example:

torch.cat(tensors, dim=0, out=None)

```

## Parameters

> - **tensors** (sequence of Tensors) – The tensors to concatenate.
> - **dim** (int) – The dimension along which the tensors are concatenated.

Reviewer: ```suggestion
- `tensors`: A sequence of tensors to be concatenated. All tensors must have the same shape in all dimensions except for the specified dimension.
- `dim`: An integer specifying the dimension along which the tensors will be concatenated. The default value is 0, which means concatenation will occur along the first dimension.
```

Reviewer: if you add the `out` in syntax, please do add it here as well :)

> ## Example

Reviewer: ```suggestion
## Example 1
```

> ## 1. Concatenating tensors along the first dimension

Reviewer: ```suggestion
The example below showcases concatenating tensors along the first dimension using the `.cat()` function:
```

```py
import torch

# Create two tensors of shape (2, 3)
tensor1 = torch.tensor([[1, 2, 3], [4, 5, 6]])
tensor2 = torch.tensor([[7, 8, 9], [10, 11, 12]])

# Concatenate the tensors along the first dimension
result = torch.cat((tensor1, tensor2), dim=0)

print("tensor1:")
print(tensor1)

print("\ntensor2:")
print(tensor2)

print("\nresult:")
print(result)
```

The output shows the two tensors and the concatenated tensor along the first dimension:

```shell
tensor1:
tensor([[1, 2, 3],
        [4, 5, 6]])

tensor2:
tensor([[ 7,  8,  9],
        [10, 11, 12]])

result:
tensor([[ 1,  2,  3],
        [ 4,  5,  6],
        [ 7,  8,  9],
        [10, 11, 12]])
```

> ## 2. Concatenating tensors along the second dimension

Reviewer: ```suggestion
## Example 2

The example below showcases concatenating tensors along the second dimension using the `.cat()` function:
```

```py
import torch

# Create two tensors of shape (2, 3)
tensor1 = torch.tensor([[1, 2, 3], [4, 5, 6]])
tensor2 = torch.tensor([[7, 8, 9], [10, 11, 12]])

# Concatenate the tensors along the second dimension
result = torch.cat((tensor1, tensor2), dim=1)

print("tensor1:")
print(tensor1)

print("\ntensor2:")
print(tensor2)

print("\nresult:")
print(result)
```

The output shows the two tensors and the concatenated tensor along the second dimension:

```shell
tensor1:
tensor([[1, 2, 3],
        [4, 5, 6]])

tensor2:
tensor([[ 7,  8,  9],
        [10, 11, 12]])

result:
tensor([[ 1,  2,  3,  7,  8,  9],
        [ 4,  5,  6, 10, 11, 12]])
```

> ## 3. Concatenating tensors along the third dimension

Reviewer: ```suggestion
## Example 3

The example below showcases concatenating tensors along the third dimension using the `.cat()` function:
```
